curator-x-discovery/src/main/java/org/apache/curator/x/discovery/details/ServiceCacheImpl.java

```diff
@@ -163,9 +164,19 @@ public void removeListener(ServiceCacheListener listener)
         discovery.getClient().getConnectionStateListenable().removeListener(listener);
     }
 
+    private boolean isRootPathEvent(PathChildrenCacheEvent event) {
```

Comment:
I would rather call it something like **isEmptyPathEvent**, because "root" is a relative thing.